File: utils/TreeNodeLeetCodeParse.go
package utils

import (
	"strings"

	treenode "github.com/masx200/leetcode-TreeNode-go"
	serialize_and_deserialize_binary_tree "github.com/masx200/leetcode-test/serialize-and-deserialize-binary-tree"
)

type TreeNode = serialize_and_deserialize_binary_tree.TreeNode

func TreeNodeLeetCodeParse(s string) *TreeNode {
	s = strings.ReplaceAll(s, " ", "")
	var t, e = treenode.NewTreeNode(s)
	if e != nil {
		panic(e)
	}
	return treeNodeNew(t)
}

func treeNodeNew(t *treenode.TreeNode) *TreeNode {

	if t == nil {
		return nil
	}
	return &TreeNode{Val: t.Val, Left: treeNodeNew(t.Left), Right: treeNodeNew(t.Right)}
}

File: utils/TreeNodeLeetCodeParse.kt
package com.github.masx200.leetcode_test.utils

import com.github.masx200.leetcode_test.insert_into_a_binary_search_tree.TreeNode
import com.github.masx200.leetcode_treenode_java.TreeNode.constructTree
import kotlinx.serialization.decodeFromString
import kotlinx.serialization.json.Json

fun TreeNodeLeetCodeParse(s: String): TreeNode? {
    return treeNodeNew(constructTree(Json.decodeFromString<Array<Int?>>(s)))
}
typealias TreeNodeClass = com.github.masx200.leetcode_treenode_java.TreeNode

private fun treeNodeNew(t: TreeNodeClass?): TreeNode? {
    if (t == null) {
        return null
    }
    val n = TreeNode()
    n.`val` = t.`val`
    n.left = treeNodeNew(t.left)
    n.right = treeNodeNew(t.right)
    return n
}